spec/services/slack_notification_service_spec.rb

```diff
@@ -0,0 +1,78 @@
+require 'rails_helper'
+
+describe SlackNotificationService do
+  describe 'initialization' do
+    subject { described_class.new({}) }
+
+    it 'raise an error if no action is set in params' do
+      expect { subject }.to raise_error(ArgumentError)
+    end
+  end
+
+  describe '#send_notification' do
+    subject { described_class.new(params).send_notification }
```

Comment:
I am not sure about this. Isn't it better if we only create the object in the "subject", and after call the method? 🤔

Reply:
not sure what style is better. The subject under test in this scenario is the method